src/serialization.rs

```diff
@@ -1,7 +1,9 @@
-use crate::{
-    AffinePoint, Bls12377, Bls12377Base, Bls12377Scalar, Curve, Field, Tweedledee, TweedledeeBase,
-    Tweedledum, TweedledumBase,
-};
+use crate::{AffinePoint, Curve, Field};
+use serde::de::Error as DeError;
+use serde::de::Visitor;
+use serde::ser::Error as SerdeError;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt;
 use std::io::{Error, ErrorKind, Read, Result, Write};
 
 pub trait ToBytes {
```

Comment:
Would it remove sense to remove these now that we use serde's `Serialize`/`Deserialize`?

A slight complication is that I don't think we could `impl<F: Field> Serialize for F` due to coherence rules, but I think it would be reasonable to make `Serialize`/`Deserialize` supertraits of `Field`.

Reply:
So `AffineCurve` internally uses `To/FromBytes` to implement `(De)Serialize`. `Field` should too and I'm going to fix this now. I think it makes sense to keep these traits to have a canonical way to read/write encoded field elements / curve points without relying on Serde. But if you think it'd be cleaner to remove them, I can do that.

Reply:
Oh ok, sounds good for now. In the long term I think we might end up with different serialization methods for perf reasons, particularly for Montgomery vs canonical field impls. (I think Hamish is planning to switch from Montgomery to canonical for the Tweedle curves.)